modules/hpa/main.tf

```diff
@@ -0,0 +1,757 @@
+            action: keep
+          - source_labels:
+              - namespace
+            regex: (sn-system|kube-system|olm|cert-manager)
+            action: drop
+      EOT
+    }
+    "type" = "Opaque"
+  }
+}
+
+resource "kubernetes_manifest" "prometheus_sn_system_scaling_prometheus" {
+  manifest = {
+    "apiVersion" = "monitoring.coreos.com/v1"
+    "kind" = "Prometheus"
+    "metadata" = {
+      "name" = "scaling-prometheus"
+      "namespace" = "sn-system"
+    }
+    "spec" = {
+      "additionalScrapeConfigs" = {
+        "key" = "scrape.yaml"
+        "name" = "scaling-conf"
+      }
+      "evaluationInterval" = var.scaling_prometheus_evaluation_interval
+      "image" = "prom/prometheus"
+      "podMetadata" = {
+        "labels" = {
+          "app" = "prometheus"
+          "cloud.streamnative.io/role" = "scaling-prometheus"
+        }
+      }
+      "replicas" = var.scaling_prometheus_replicas
+      "resources" = {
+        "limits" = {
+          "cpu" = var.scaling_prometheus_cpu_limit
+          "memory" = var.scaling_prometheus_memory_limit
+        }
+      }
+      "retention" = var.scaling_prometheus_retention_period
+      "scrapeInterval" = var.scaling_prometheus_scrape_interval
+      "securityContext" = {
+        "fsGroup" = 2000
+        "runAsNonRoot" = true
+        "runAsUser" = 1000
+      }
+      "serviceAccountName" = "scaling-prometheus"
+      "serviceMonitorSelector" = {
+        "matchLabels" = {
+          "app" = "prometheus"
+        }
+      }
+      "version" = var.scaling_prometheus_version
+    }
+  }
+}
+
+resource "kubernetes_manifest" "service_sn_system_scaling_prometheus" {
+  manifest = {
+    "apiVersion" = "v1"
+    "kind" = "Service"
+    "metadata" = {
+      "name" = "scaling-prometheus"
+      "namespace" = "sn-system"
+    }
+    "spec" = {
+      "ports" = [
+        {
+          "port" = 9090
+          "protocol" = "TCP"
+          "targetPort" = 9090
+        },
+      ]
+      "selector" = {
+        "cloud.streamnative.io/role" = "scaling-prometheus"
+      }
+    }
+  }
+}
+
+
+# generate a private key for internal issuer's cert
+resource "tls_private_key" "private_key" {
+  algorithm   = "RSA"
+}
+
+# generate a self-signed cert for internal issuer
+resource "tls_self_signed_cert" "issuer_cert" {
+  key_algorithm   = "RSA"
+  private_key_pem = tls_private_key.private_key.private_key_pem
+
+  subject {
+    common_name  = "CA Issuer"
+  }
+
+  validity_period_hours = 87600
+
+  is_ca_certificate = true
+
+  allowed_uses = [
+    "key_encipherment",
+    "digital_signature",
+    "cert_signing",
+  ]
+
+  depends_on = [
+    tls_private_key.private_key
+  ]
+}
+
+resource "kubernetes_manifest" "secret_issuer_cert" {
+  computed_fields = ["stringData"]
+  manifest = {
+    "apiVersion" = "v1"
+    "kind" = "Secret"
+    "metadata" = {
+      "name" = "issuer-cert"
+      "namespace" = "cert-manager"
```

Comment:
We're not currently using this namespace. Should we be putting cert-manager into its own dedicated namespace?

Reply:
I see we're using sn's namespace for cert-manager, just added a dedicated var for it.